### PR TITLE
Reduce usage of word kill

### LINKED ISSUE #7858: Remove "kill" from examples
We have deprecated "suicide" from the Yellow Paper, so we can remove "kill" from the examples and documentation.

### PR DESCRIPTION
**Pull request by fulldecent**
Reduces the usage of the word kill, favoring other words where possible

Fixes #7858.

### CODE REVIEW

File: test/tools/ossfuzz/config/solidity.dict [hunk 1]
```diff
@@ -40,7 +40,7 @@
 ".creationCode"
 ".delegatecall("
 ".gas("
-".kill("
```
Comment (axic) [inline test/tools/ossfuzz/config/solidity.dict:43]: @bshastry can you check this?

Comment (bshastry) [reply test/tools/ossfuzz/config/solidity.dict:43]: Question: Is the issue that using "kill" is illegal? Or is it that we should discourage developers from using it? If it's the latter I don't see why we should remove it from the dictionary list. The dictionary list is meant to make it easier for the fuzzer to synthesize corner cases, so as long as "kill" is legal, it is fine to keep it.

Comment (fulldecent) [reply test/tools/ossfuzz/config/solidity.dict:43]: Updated at 4476625456bce83aee1cab31c2d274bd65c0ec3d

**Comment by chriseth** [comment]
I don't think it is necessary to rename internal functions for now. Furthermore, we should also change the name "mortal" because it does not fit the pattern anymore.

**Comment by fulldecent** [comment]
* Renamed mortal
* Removed changes to internal code
* Squashed

**Comment by chriseth** [comment]
Maybe it's because I'm not a native speaker, but "retrievable" does not really resonate with the concept in my head...

**Comment by fulldecent** [comment]
Other ideas:

* Wind Up/`windupable`. 
* `surrogate`
* `temporary`
* `destructible`
* `ephemeral`

**Comment by chriseth** [comment]
I think I like destructible most

**Comment by ChrisChinchilla** [comment]
Destructible isn't my favourite option as it still feels a bit "strong", but it's fine. My vote would go for ephemeral (maybe too complex), or temporary.

**Comment by ChrisChinchilla** [comment]
Rest looks good, thanks @fulldecent !

**Comment by bshastry** [comment]
This PR needs rebasing.

**Comment by chriseth** [comment]
Please use rebase instead of merge.

**Comment by Marenz** [comment]
I rebased this and changed `Retrievable` to `Temporary`

**Comment by bshastry** [comment]
Please rebase once again, that will most likely fix the failing osx build

**Comment by Marenz** [comment]
pushed

**Review by bshastry — APPROVED** [review]
LGTM

**Comment by chriseth** [comment]
The change was not consistent. Also, temporary sounds to me that the contract will be temporary for sure, while we actually only provide the option to make it non-permanent.

File: docs/contracts/function-modifiers.rst [hunk 2]
```diff
@@ -39,7 +39,7 @@ if they are marked ``virtual``. For details, please see
         }
     }
 
-    contract mortal is owned {
```
Comment (chriseth) [inline docs/contracts/function-modifiers.rst:42]: This should have resulted in a test failure, because the contract is referenced below without change.

Comment (erak) [reply docs/contracts/function-modifiers.rst:42]: Missing test runs usually result from wrong code indentation and should have been fixed by https://github.com/ethereum/solidity/pull/7564. Perhaps we should check it again.

Comment (chriseth) [reply docs/contracts/function-modifiers.rst:42]: Created an issue about it.

Comment (Marenz) [reply docs/contracts/function-modifiers.rst:42]: it is? I can't find `mortal` anywhere in the whole repo?

Comment (chriseth) [reply docs/contracts/function-modifiers.rst:42]: You are right, it was weirdly inheriting from `owned`...: https://github.com/ethereum/solidity/pull/8111/files#diff-2e640333fd552416d91547f68fb74857L61

**Pull request merged**